app/models/smart_proxy.rb

```diff
@@ -88,12 +86,20 @@ def has_feature?(feature)
     self.features.any? { |proxy_feature| proxy_feature.name == feature }
   end
 
-  private
-
-  def proxy_status
-    @proxy_status ||= ProxyStatus.new(self)
+  def status
+    return @status if @status
+    @status = { :version => ProxyStatus::Version.new(self) }
+    features.map(&:name).each do |feature|
+      normalized_feature_name = feature.delete(' ')
+      status = ProxyStatus.find_status_by_humanized_name(normalized_feature_name)
```

Comment:
nitpick, 'status' here as a local variable & as a instance variable `@status` can be confusing to newcomers, perhaps `feature_status` is a little less confusing. `status_symbol` would then become `feature_status_symbol`